### PR TITLE
Add iterate method for Root objects

### PR DESCRIPTION
**Pull request by edtrelo**
Solves #180

### CODE REVIEW

**Comment by Kolaru** [comment]
Thanks for the contribution.

If you have any idea why the x86 tests are failing, that would be great, I'm personally quite confused.

File: src/root_object.jl [hunk 1]
```diff
@@ -50,3 +50,42 @@ show(io::IO, rt::Root) = print(io, "Root($(rt.interval), :$(rt.status))")
...
+    - `r`: Root object.
+    - `state`: indicates the prior element iterated.
+
+    Outputs:
+    - `r.status`: status of the root.
+    - `state`: indicates that the second iterable element of r has been returned.
+    or 
+    - nothing: the iteration has been completed.
+"""
+function Base.iterate(r::Root{T}, state::Int64) where {T}
```
Comment (lbenet) [inline src/root_object.jl:84]: Thanks @edtrelo for your contribution.

Following on [this comment](https://github.com/JuliaIntervals/IntervalRootFinding.jl/pull/190#issuecomment-1591086103), I think the error is due to this line: it imposes `state` to be `Int64`. I think the problem is solved if `state` is required to be `Integer`

Comment (edtrelo) [reply src/root_object.jl]: Thanks for the sugguestion @lbenet. I've alredy changed the type of `state`.

**Comment by Kolaru** [comment]
Absolutely. I will look into it soon (tm).

**Comment by Kolaru** [comment]
I've reformatted things a bit and rebased it. Thanks a lot for your work @edtrelo and sorry for the long delay

**Pull request merged**